lib/TextDocument/TextDocumentUri.php

```diff
@@ -82,7 +85,10 @@ public static function fromString(?string $uri): self
             ));
         }
 
-        $path = Path::makeAbsolute(substr($path, 1), '/');
+        if (str_starts_with($path, '/')) {
+            $path = substr($path, 1);
+        }
```

Comment:
would `ltrim($path, '/')` be better?

Reply:
I think not, because there can be more than one `/` in front... We do have tests tho, so I'll try :)

The problem is that PHP is using URIs in a non-standard way. The correct form is:
`schema:// (SERVER) (/path)`. When there is no server, this looks like `schema://(/path)`. Or for relative path `schema://(path)`.
In phar case we have `phar://(/path)` on Unix but `phar://c:/path` on Windows...

What a mess...